### PR TITLE
[Model] Let `QKVParallelLinear` de-interleave fused QKV weights

### PR DESCRIPTION
**Pull request by hmellor**
Part of #48972, extracted from #49038.

Several models store their fused QKV projection interleaved per KV group on disk (`[q_0..q_{g-1}, k, v]` repeated per KV head) rather than block contiguous (`[Q|K|V]`). Each of them re-implemented the same reordering in `load_weights`.

This adds `fused_qkv_interleaved` to `QKVParallelLinear`, so the layer de-interleaves the weight itself in the two paths that handle a checkpoint that is already fused (`_load_fused_module_from_checkpoint`, and `weight_loader` with `loaded_shard_id=None`). The de-interleave respects the parameter's `output_dim`, preserving Falcon's handling of quantised checkpoints where the output dim is 1.

Bloom, Falcon and GPT-NeoX now set the flag and drop their bespoke repacking. For Falcon that removes the entire hand written `FalconModel.load_weights`, since the rest of it duplicated `AutoWeightsLoader`:

- the GPTQ extra bias skip is already covered by `AutoWeightsLoader.load_weights`, which appends `.bias` to `ignore_unexpected_suffixes`
- the `is_pp_missing_parameter` check is already covered by `_load_module` returning early for `PPMissingLayer`/`StageMissingLayer`
- the `getattr(param, "weight_loader", default_weight_loader)` dispatch is already `_load_param`

`BloomModel.load_weights` is removed for the same reason: once `_repack_qkv` is gone it is a bare `AutoWeightsLoader(self)` wrapper, which is what the parent's recursion does anyway. `GPTNeoXModel.load_weights` stays because it still carries `skip_substrs`, which is #53106's territory.

### Not a duplicate

The only open PR touching this is #49038, which is mine and which this is extracted from. #49038 is large and mixes this with the `autoload_weights` infrastructure; splitting it out lets the QKV change be reviewed and land on its own. I will rebase #49038 on top of this. A search for other open PRs touching Falcon/Bloom/GPT-NeoX weight loading or `QKVParallelLinear` turned up nothing overlapping.

### Testing

New unit test asserting an interleaved fused checkpoint loads to the same `[Q|K|V]` as separate Q/K/V weights, over MHA/GQA/MQA head counts and over both weight and bias:

```
$ pytest tests/model_executor/layers/test_qkv_parallel_linear.py -q
6 passed
```

The test discriminates: flipping `fused_qkv_interleaved` to `False` fails 4 of the 6 cases (the two MQA cases are no-ops by construction, since one KV head means the interleaved layout already equals `[Q|K|V]`).

The new `_deinterleave_fused_qkv` was also checked to be bit-exact against both removed implementations (`torch.equal`) across MHA/GQA/MQA shapes, 2-D weights with `output_dim` 0 and 1, and 1-D biases.

Each affected architecture was then built on CPU and driven through its real `load_weights` with a synthetic HF-layout checkpoint, asserting every parameter is consumed and that the fused QKV tensors land exactly as `torch.cat([q, k, v])`:

| Model | Config | Result |
| --- | --- | --- |
| `tiiuae/falcon-rw-1b` | MHA, 32 heads | 27/27 params, qkv exact |
| `tiiuae/falcon-7b` | MQA, 71 heads / 1 KV head | 16/16 params, qkv exact |
| `tiiuae/falcon-40b` | new decoder arch, 128 heads / 8 KV heads | 20/20 params, qkv exact |
| `bigscience/bloom-560m` | 16 heads | 29/29 params, qkv weight + bias exact |
| `EleutherAI/pythia-70m` | 8 heads | 28/28 params, qkv weight + bias exact |

That covers all three Falcon branches the deleted code special cased.

### Model evaluation

Correctness here is a bit-exact tensor layout property, and it is verified above against the exact code being removed, on every affected architecture. End to end generation evals have not been run yet as I do not have GPU hardware to hand right now; I will post results for Bloom, Falcon and GPT-NeoX before merge.

### AI assistance

AI assistance was used for this change. I have reviewed every changed line and understand and stand behind the change.

### CODE REVIEW

File: vllm/model_executor/models/bloom.py [hunk 1]
```diff
@@ -295,24 +296,6 @@ def forward(
...
-        # BLOOM's fused QKV is laid out as (num_heads * 3 * head_size) on its
-        # output dim (0), while vLLM expects (3 * num_heads * head_size).
-        num_heads = self.config.num_attention_heads
-        for name, loaded_weight in weights:
-            if "query_key_value" in name:
-                shape = loaded_weight.shape
-                loaded_weight = loaded_weight.view((num_heads, 3, -1) + shape[1:])
-                loaded_weight = loaded_weight.transpose(0, 1)
-                loaded_weight = loaded_weight.reshape(shape)
-            yield name, loaded_weight
```
Comment (Isotr0py) [inline vllm/model_executor/models/bloom.py:310]: Usually, for this kind of qkv layout, we split it in attention runtime, otherwise it's quite difficult to deal with quantization + TP:
https://github.com/vllm-project/vllm/blob/de216b6e6487b8b452805b582b242616ee616682/vllm/model_executor/models/qwen2_vl.py#L326-L390
I think we missed update for these models because they're super old. 😅

**Comment by hmellor** [comment]
Closing as this is not the right fix and these models are not important enough for such an abstraction
